larreco/HitFinder/HitFinderTools/PeakFitterGaussian_tool.cc

```diff
@@ -270,13 +268,222 @@ namespace reco_tool {
         parIdx += 3;
       }
 
+      if (fRefit && chi2PerNDF > fRefitThreshold &&
+          chi2PerNDF < std::numeric_limits<double>::infinity()) {
+
+        const ICandidateHitFinder::HitCandidate refitParams =
+          FindRefitCand(Gaus, roiSignalVec, startTime, roiSize, hitCandidateVec, peakParamsVec);
+
+        if (refitParams.hitHeight > 0) {
+
+          ICandidateHitFinder::HitCandidateVec newHitCandidateVec = hitCandidateVec;
+          newHitCandidateVec.push_back(refitParams);
+
+          // Build the string to describe the fit formula
+          unsigned int const nGausNew = nGaus + 1;
+          assert(fFitCache.Get(nGausNew));
+          Gaus = *(fFitCache.Get(nGausNew));
```

Comment:
Unless you need to cache `Gaus`, I would avoid the copy and use a reference:

```C++
auto& GausNew = *fFitCache.Get(nGausNew);
```

and then there is no need to remove the reference on line 187.

Reply:
Initially I tried using a reference, however found that the second fit call often failed to converge. I found that copying the Gaussians prevented this failure, although I could never get to the root cause of the issue. I figured that the overhead of copying the function would be negligible in comparison to the time required to run the fits so it was not the end of the world if we ended up doing an extra copy.